auth/opa/rpcauth/input.go

```diff
@@ -54,6 +54,10 @@ type RPCAuthInput struct {
 	// Information about the host serving the RPC.
 	Host *HostAuthInput `json:"host"`
 
+	// Whether this RPCAuthInput was generated via a proxy
+	// or directly in the RPC interceptor.
+	FromProxy bool `json:"from_proxy"`
```

Comment:
optional:

Assuming we can't stuff this into 'Extensions', it might be worth stuffing this into a more general field that could capture other things, rather than as a top-level.

Something like:


```go
// EnvironmentInput contains information about the environment in which the policy evaluation is
// happening.
type EnvironmentInput struct {
   // True if the policy evaluation is being performed by some entity other than the host
   // (for example, policy checks performed by an authorizing proxy or other agent)
   NonHostPolicyCheck bool `json:"non_host_policy_check"`
}
```

Reply:
Good point. I like this. Integrated